pkg/controllers/dashboardapi/settings/settings_test.go

```diff
@@ -0,0 +1,129 @@
+package settings
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type testCase struct {
+	description     string
+	envVar          *string
+	newSetting      settings.Setting
+	existingSetting *v3.Setting
```

Comment:
One suggestion here - I've found it helpful to include the expected outcome in test cases. That way it's easier for people to see what you are looking to get out of a test case and how to make a new one if the code changes. You don't have to make this change, just a suggestion.

Reply:
I already have description for the test case so I went ahead changed the description to hopefully explain what the expected behavior is of each.